### ISSUE TITLE
When starting a new building task, the old one does not stop

### ISSUE THREAD

**Issue by chazeon** [James-Yu/LaTeX-Workshop#140]
## Problem
For current release, when starting a new building task before the previous one is done, the new one simply runs without killing the previous job or cleaning the aux files. The simultaneous running tasks causes two working task writing to same series of files, and these files is guaranteed to be broken. This could be especically bothering when, the build on save feature is enabled, which by default it is so, that I have to stop the previous toolchain manually by killing the running task in the Task Manager. This could go even crazy when:
* the entire project took a long time to build
* one have to make adjustment on the main file and included file in a same modification

## Possible Solutions
I haven't thought out a complete solution for this problem, but I could give some suggestions for consideration:
1. This auto build feature could use something like "debounce", like someting in the lodash library: only call the build once in a while, and the interval could be set in the settings, or simply put the newly assigned build job on hold before the previous one is finish.
2. Make a quick switch, from the command palette, say quick toggle for auto build or something like this. It is uninviting for users to modify the settings once in the while.
3. Kill the previous task and clean the aux files - the quick and dirty way.
4. Make another option like "save all and build" and users could use this hot key when one think the file is ready for build.
5. Change the "Build LaTeX project" to "Stop building" when a task is running (not sure it is possible) or simply add a "Stop..." entry to the menu.

The bottom line is that: saving a file should not be an "expensive" job, even when the auto build feature is enabled; and one should not need to keep a Task Manager open when writing a LaTeX file.

I personally disabled this automation feature for now, and keep a Task Manager running.

**Comment by James-Yu** [comment]
Thanks for the detailed issue. I will look into it.

**Comment by James-Yu** [comment]
Even in current version, `BUILD` tries to kill the process before starting a new one: https://github.com/James-Yu/LaTeX-Workshop/blob/master/src/builder.ts#L22 .
I will publish a new version with more debugging info to see what is the problem.

**Comment by chazeon** [comment]
Thanks! If there is any information I can provide, let me know.
